components/Details.js

```diff
@@ -0,0 +1,74 @@
+import { useState, useEffect } from 'react';
+
+import { transformDetailsToFields } from './../utils/transformData';
+import { FormInput } from './Form/FormInput';
+export const Details = ({ detailsDataCallback, defaultValue }) => {
+  const [detailFields, setDetailFields] = useState(
+    transformDetailsToFields(defaultValue) || [],
+  );
+
+  const handleAddFields = (event) => {
+    event.preventDefault();
+    setDetailFields([...detailFields, { fieldName: '', fieldContent: '' }]);
+  };
+
+  const handleDeleteFields = (index, event) => {
+    event.preventDefault();
+    const data = [...detailFields];
+    data.splice(index, 1);
+    setDetailFields(data);
+  };
+
+  const handleFormChange = (index, event) => {
+    const data = [...detailFields];
+    data[index][event.target.name] = event.target.value;
+    setDetailFields(data);
+  };
```

Comment:
In general, if you want to clone something not to MUTATE the original structure. you have two approaches

const newArr = [ ...oldArray ]

or 

const newArray = oldArray.slice(0)

it is equivalent

Reply:
```suggestion
  const handleFormChange = (index, { target }) => {
    const newData = newArray.map((item) => item.fieldName === target.name ? { ...item, fieldName: target.value} : item)
    
    setDetailFields(data);
  };
```

Reply:
I reused your code

```

  const handleFormChange = (index, { target }) => {
    const newData = detailFields.map((item, i) =>
      index === i ? { ...item, [target.name]: target.value } : item,
    );

    setDetailFields(newData);
  };
```